src/OpenSage.Game/Logic/Player.cs

```diff
@@ -169,6 +205,37 @@ public void DeselectUnits()
             _selectedUnits.Clear();
         }
 
+        public bool ScienceAvailable(Science science)
+        {
+            if (Sciences.Contains(science)) return false;
+
+            foreach (var requiredScience in science.PrerequisiteSciences)
+            {
+                if (requiredScience.Value == null)
+                {
+                    continue;
+                }
+                if (!Sciences.Contains(requiredScience.Value))
+                {
+                    return false;
+                }
+            }
```

Comment:
This could probably be expressed more easily as a LINQ query, but I'm not sure if that's recommendable as it's called from an `Update` method.